Exercise 6
Flicker

In [15]:
import numpy as np
import pandas as pd
flicker=pd.read_csv("/content/BL-Flickr-Images-Book-subset.csv",sep=",")
flicker

Identifier                                  Edition Statement  \
0            206                                                NaN   
1            216                                                NaN   
2            218                                                NaN   
3            472                                                NaN   
4            480                       A new edition, revised, etc.   
...          ...                                                ...   
2875     1413820                                                NaN   
2876     1413823  [Another edition.] With notes by Dean Milman a...   
2877     1413828                                                NaN   
2878     1413858                                                NaN   
2879     1414094                                                NaN   

          Place of Publication Date of Publication              Publisher  \
0                       London         1879 [1878]       S. Tinsley & Co.   
1     London; Virtue & Yorston                1868           Virtue & Co.   
2                       London                1869  Bradbury, Evans & Co.   
3                       London                1851          James Darling   
4                       London                1857   Wertheim & Macintosh   
...                        ...                 ...                    ...   
2875                    London                1848          Longman & Co.   
2876                    London             1854-55                    NaN   
2877                    London          [1882, 85]       Ward, Lock & Co.   
2878                    London           1899-1901            John Murray   
2879                    London                1898          Methuen & Co.   

                                                  Title  \
0                     Walter Forbes. [A novel.] By A. A   
1     All for Greed. [A novel. The dedication signed...   
2     Love the Avenger. By the author of “All for Gr...   
3     Welsh Sketches, chiefly ecclesiastical, to the...   
4     [The World in which I live, and my place in it...   
...                                                 ...   
2875  The History of the Decline and Fall of the Rom...   
2876  [The History of the Decline and Fall of the Ro...   
2877  Gibbon's History of the Decline and Fall of th...   
2878  The Student's Gibbon ... Abridged from the ori...   
2879  Exploration and Hunting in Central Africa, 189...   

                         Author  \
0                         A. A.   
1                     A., A. A.   
2                     A., A. A.   
3                     A., E. S.   
4                     A., E. S.   
...                         ...   
2875             Gibbon, Edward   
2876             Gibbon, Edward   
2877             Gibbon, Edward   
2878             Gibbon, Edward   
2879  GIBBONS, Alfred St. Hill.   

                                           Contributors  Corporate Author  \
0                                       FORBES, Walter.               NaN   
1          BLAZE DE BURY, Marie Pauline Rose - Baroness               NaN   
2          BLAZE DE BURY, Marie Pauline Rose - Baroness               NaN   
3                           Appleyard, Ernest Silvanus.               NaN   
4                                   BROOME, John Henry.               NaN   
...                                                 ...               ...   
2875                         SCHREITER, Carl Gottfried.               NaN   
2876  GUIZOT, François Pierre Guillaume.|Milman, He...               NaN   
2877             BURY, John Bagnell.|Milman, Henry Hart               NaN   
2878  ANDERSON, John George Clark.|GREENIDGE, Abel H...               NaN   
2879                                  WHYMPER, Charles.               NaN   

      Corporate Contributors  Former owner  Engraver Issuance type  \
0                        NaN           NaN       NaN   monographic   
1                        NaN     

We display the columns and the null values in them

In [16]:
flicker.isnull().sum()

Identifier                   0
Edition Statement         2575
Place of Publication         0
Date of Publication         32
Publisher                 1080
Title                        0
Author                     646
Contributors                 0
Corporate Author          2880
Corporate Contributors    2880
Former owner              2880
Engraver                  2880
Issuance type                0
Flickr URL                   0
Shelfmarks                   0
dtype: int64

Then, We delete the 4 columns that are empty 2880 nan out of 2880 rows

In [17]:
columnstodelete = ["Corporate Author", "Corporate Contributors", "Former owner", "Engraver"]
flicker  = flicker.drop(columns=columnstodelete)

We then drop the rows with null values in them in order to keep only value without nan

In [18]:
flicker = flicker.dropna()
flicker

Identifier                                  Edition Statement  \
4            480                       A new edition, revised, etc.   
5            481                      Fourth edition, revised, etc.   
24          5385      [Another edition.] Illustrated by John Leech.   
25          5389                                 [Another edition.]   
44         16544                                 [Another edition.]   
...          ...                                                ...   
2560     1230911  [Another edition.] With a memoir of the author...   
2619     1257448  Décima octava edicion. Corregida y aumentada ...   
2679     1306654  [Another edition.] In a North Country Village ...   
2795     1368095                            Third edition- revised.   
2835     1384312                                 [Another edition.]   

         Place of Publication Date of Publication  \
4                      London                1857   
5                      London                1875   
24                     London             [1897?]   
25                     London             [1897?]   
44                     Boston                1817   
...                       ...                 ...   
2560  George Routledge & Sons                1895   
2619                   Madrid                1854   
2679                   London                1896   
2795                 Montreal                1874   
2835                   London              [1897]   

                                              Publisher  \
4                                  Wertheim & Macintosh   
5                                     William Macintosh   
24                                Bradbury, Agnew & Co.   
25                                Bradbury, Agnew & Co.   
44                                       Wells & Lilley   
...                                                 ...   
2560                                             London   
2619  Officinas de Biblioteca Universal de Autores C...   
2679                                       Osgood & Co.   
2795                                          J. Lovell   
2835                                      Cassell & Co.   

                                                  Title  \
4     [The World in which I live, and my place in it...   
5     [The World in which I live, and my place in it...   
24    [The comic history of England ... With twenty ...   
25    [The Comic History of Rome ... Illustrated by ...   
44    [The Narrative of Robert Adams, a sailor, who ...   
...                                                 ...   
2560  [The history of Tom Jones, a foundling.]: Tom ...   
2619  [Clave Historial con que se abre la puerta a l...   
2679                      [In a North Country Village.]   
2795  [History of Canada ... Translated ... and acco...   
2835  [The Holy Land and the Bible. A book of Script...   

                                             Author  \
4                                         A., E. S.   
5                                         A., E. S.   
24                       A'BECKETT, Gilbert Abbott.   
25                       A'BECKETT, Gilbert Abbott.   
44                           ADAMS, Robert - Sailor   
...                                             ...   
2560                                Fielding, Henry   
2619                              FLOREZ, Henrique.   
2679  FRANCIS, M. E. - pseud. [i.e. Mary Blundell.]   
2795                     GARNEAU, François Xavier.   
2835                       GEIKIE, John Cunningham.   

                            Contributors Issuance type  \
4                    BROOME, John Henry.   monographic   
5                    BROOME, John Henry.   monographic   
24                  LEECH, John - Artist   monographic   
25                  LEECH, John - Artist   monographic   
44                          COCK, Simon.   monographic   
...                                  ...           ...   
2560   T., G. H.|TOWNSEND, Geor

Now, we need to make sure that the data is usable in the publication date column, that is to say that all the data is of the datetime type. We ensure that there will be no errors by using the error constraint.



In [19]:
flicker['Date of Publication'] = pd.to_datetime(flicker["Date of Publication"], errors='coerce')

Then, we display the information about the cleaned dataset (only 143 value over 190 are datetime type in the publication data column).

In [20]:
flicker.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 190 entries, 4 to 2835
Data columns (total 11 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   Identifier            190 non-null    int64         
 1   Edition Statement     190 non-null    object        
 2   Place of Publication  190 non-null    object        
 3   Date of Publication   143 non-null    datetime64[ns]
 4   Publisher             190 non-null    object        
 5   Title                 190 non-null    object        
 6   Author                190 non-null    object        
 7   Contributors          190 non-null    object        
 8   Issuance type         190 non-null    object        
 9   Flickr URL            190 non-null    object        
 10  Shelfmarks            190 non-null    object        
dtypes: datetime64[ns](1), int64(1), object(9)
memory usage: 17.8+ KB


Finally, we display a summary statistics of the date of publication column

In [21]:
flicker['Date of Publication'].describe()

<ipython-input-21-cac9305613a7>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  flicker['Date of Publication'].describe()


count                     143
unique                     76
top       1896-01-01 00:00:00
freq                        6
first     1720-01-01 00:00:00
last      1899-01-01 00:00:00
Name: Date of Publication, dtype: object

The data cleaned is :

In [22]:
flicker

Identifier                                  Edition Statement  \
4            480                       A new edition, revised, etc.   
5            481                      Fourth edition, revised, etc.   
24          5385      [Another edition.] Illustrated by John Leech.   
25          5389                                 [Another edition.]   
44         16544                                 [Another edition.]   
...          ...                                                ...   
2560     1230911  [Another edition.] With a memoir of the author...   
2619     1257448  Décima octava edicion. Corregida y aumentada ...   
2679     1306654  [Another edition.] In a North Country Village ...   
2795     1368095                            Third edition- revised.   
2835     1384312                                 [Another edition.]   

         Place of Publication Date of Publication  \
4                      London          1857-01-01   
5                      London          1875-01-01   
24                     London                 NaT   
25                     London                 NaT   
44                     Boston          1817-01-01   
...                       ...                 ...   
2560  George Routledge & Sons          1895-01-01   
2619                   Madrid          1854-01-01   
2679                   London          1896-01-01   
2795                 Montreal          1874-01-01   
2835                   London                 NaT   

                                              Publisher  \
4                                  Wertheim & Macintosh   
5                                     William Macintosh   
24                                Bradbury, Agnew & Co.   
25                                Bradbury, Agnew & Co.   
44                                       Wells & Lilley   
...                                                 ...   
2560                                             London   
2619  Officinas de Biblioteca Universal de Autores C...   
2679                                       Osgood & Co.   
2795                                          J. Lovell   
2835                                      Cassell & Co.   

                                                  Title  \
4     [The World in which I live, and my place in it...   
5     [The World in which I live, and my place in it...   
24    [The comic history of England ... With twenty ...   
25    [The Comic History of Rome ... Illustrated by ...   
44    [The Narrative of Robert Adams, a sailor, who ...   
...                                                 ...   
2560  [The history of Tom Jones, a foundling.]: Tom ...   
2619  [Clave Historial con que se abre la puerta a l...   
2679                      [In a North Country Village.]   
2795  [History of Canada ... Translated ... and acco...   
2835  [The Holy Land and the Bible. A book of Script...   

                                             Author  \
4                                         A., E. S.   
5                                         A., E. S.   
24                       A'BECKETT, Gilbert Abbott.   
25                       A'BECKETT, Gilbert Abbott.   
44                           ADAMS, Robert - Sailor   
...                                             ...   
2560                                Fielding, Henry   
2619                              FLOREZ, Henrique.   
2679  FRANCIS, M. E. - pseud. [i.e. Mary Blundell.]   
2795                     GARNEAU, François Xavier.   
2835                       GEIKIE, John Cunningham.   

                            Contributors Issuance type  \
4                    BROOME, John Henry.   monographic   
5                    BROOME, John Henry.   monographic   
24                  LEECH, John - Artist   monographic   
25                  LEECH, John - Artist   monographic   
44                          COCK, Simon.   monographic   
...                                  ...           ...   
2560   T., G. H.|TOWNSEND, Geor